~={red}(1point)=~ for Classify taxonomy
Visualize the taxonomy of your ASVs: (~={red}(1point)=~
Filter mito, chloro, chloro genome and include class level and below ~={red}(1point)=~
Filtered Taxa Bar Plot Questions ~={red}(10 points)=~
Phylogenetic tree ~={red}(1point)=~
slurm email output ~={red}(1point)=~

~={orange}TOTAL POINT = 15=~

ANSWERS: 
Load qiime2 in a terminal session after you go into the taxonomy folder

```
#insert the two commands to activate qiime2
module purge
module load qiime2/2024.10_amplicon
```
## Remove long (300+ base pair) amplicons from the representative sequences file and the feature table

```
# filter out any large amplicons from the seqs and table (because they are contaminates)

cd /scratch/alpine/$USER/cow/dada2

qiime feature-table filter-seqs \
    --i-data seqs_cow.qza \
    --m-metadata-file seqs_cow.qza \
    --p-where 'length(sequence) < 300' \
    --o-filtered-data cow_seqs_dada2_filtered300.qza

qiime feature-table tabulate-seqs \
--i-data cow_seqs_dada2_filtered300.qza \
--o-visualization cow_seqs_dada2_filtered300.qzv

qiime feature-table filter-features \
--i-table table_cow.qza \
--m-metadata-file cow_seqs_dada2_filtered300.qza \
--o-filtered-table cow_table_dada2_filtered300.qza
  
qiime feature-table summarize \
--i-table cow_table_dada2_filtered300.qza \
--m-sample-metadata-file ../metadata/cow_metadata.txt \
--o-visualization cow_table_dada2_filtered300.qzv
    
```

## Classify taxonomy using GreenGenes2

* First get the GreenGenes database:
```
cd /scratch/alpine/$USER/cow/taxonomy

wget --no-check-certificate https://ftp.microbio.me/greengenes_release/2024.09/2024.09.backbone.v4.nb.qza
```

- classify the ASVs (takes about 5 mins)
```
qiime feature-classifier classify-sklearn \
--i-reads ../dada2/cow_seqs_dada2_filtered300.qza \
--i-classifier 2024.09.backbone.v4.nb.qza \
--o-classification taxonomy_gg2_filtered.qza
```

- visualize the taxonomy of all the ASVs
```
qiime metadata tabulate \
--m-input-file taxonomy_gg2_filtered.qza \
--o-visualization taxonomy_gg2_filtered.qzv
```

- Filter mitochondria and chloroplast out to generate a filtered feature table, keep only ASVs with a class or lower taxonomy.
```
qiime taxa filter-table \
--i-table ../dada2/cow_table_dada2_filtered300.qza \
--i-taxonomy taxonomy_gg2_filtered.qza \
--p-exclude mitochondria,chloroplast,sp004296775 \
--p-include c__ \
--o-filtered-table ../dada2/table_nomitochloro_gg2_filtered300.qza
```

- Visualize the taxa bar plot
```
qiime taxa barplot \
--i-table ../dada2/table_nomitochloro_gg2_filtered300.qza \
--i-taxonomy taxonomy_gg2_filtered.qza \
--m-metadata-file ../metadata/cow_metadata.txt \
--o-visualization ../taxaplots/taxa_barplot_nomitochloro_gg2_filtered300.qzv
```

### Filtered Taxa Bar Plot Questions

**Question 1**: Attach a picture of your taxa bar plot, organized by cow sampling location (body_site) at the level 7 taxonomic level. What general trends do you notice?  **composition looks different by body sites.**

**_Question 2**: What are the top 2 most abundant bacterial **classes** in the fecal samples? **Clostridia_258483 and Bacteroidia**_

**_Question 3**: What highly abundant ASV is shared between both the udder and skin samples? **d__Bacteria;p__Bacillota_A_368345;c__Clostridia_258483;o__Oscillospirales;f__Oscillospiraceae_88309;g__Faecousia;s__Faecousia sp000434635**_

**_Question 4**: Which samples (still sorted by body_site) have higher alpha diversity in terms of observed features? **Will accept fecal or skin or udder or all as an answer

**Question 5**: do all samples contain archaea as well? No, not all samples contain archaea

**Question 6**: why do we filter out sp004296775? because it is a chloroplast genome

**Question 7**: what is the difference between these two flags? 
--p-exclude mitochondria,chloroplast,sp004296775 \
--p-include c__ \

one filters things out, one only includes features that has a class or lower taxonomy

**Question 8**: do the positive controls look the same as each other? 

**Question 9**: Do the negative/extraction controls (Samples labeled as EC), look like the positive controls? Yes or no? No 

**Question 10**: do the negative/extraction controls (Samples labeled as EC), look like the real samples? Yes or no? No 

## Phylogenetic tree

Create a job script to run the phylogenetic tree building. Remember you must start a new terminal session, navigate to your slurm directory, and then submit the job. You do NOT need to start any other interactive sessions.This job will take about an hour. 

Go to OnDemand and create a new text file, name it: tree.sh

```
tree.sh
```

- Put the commands into the file, and save it. 
```
#!/bin/bash
#SBATCH --job-name=tree
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --partition=amilan
#SBATCH --time=04:00:00
#SBATCH --mail-type=ALL
#SBATCH --mail-user=lindsval@colostate.edu
#SBATCH --output=slurm-%j.out
#SBATCH --qos=normal

#Activate qiime
#Insert the two commands you need to load qiime2
module purge
module load qiime2/2024.10_amplicon

#Get reference tree (gg2)
wget --no-check-certificate -P ../tree https://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.sepp-reference.qza

#Command
qiime fragment-insertion sepp \
--i-representative-sequences ../dada2/cow_seqs_dada2_filtered300.qza \
--i-reference-database ../tree/2022.10.backbone.sepp-reference.qza \
--o-tree ../tree/tree_gg2.qza \
--o-placements ../tree/tree_placements_gg2.qza
```

- submit the job from the terminal
```
 sbatch tree.sh
```
Submitted batch job 24289371

We will use this file in the next homework!

### Once this job finishes, copy and paste what the slurm email says here ~={red}(1 point)=~: 

#### for example: 
